build-logic/build.gradle

```diff
@@ -0,0 +1,42 @@
+plugins {
+  alias(libs.plugins.kotlin.jvm)
+  alias(libs.plugins.ktlint)
+  id 'java-gradle-plugin'
+}
+
+gradlePlugin {
+  plugins {
+    module {
+      id = "library"
```

Comment:
"library" is so generic, why not `com.squareup.anvil.library`?

Reply:
In my head, I consider local convention plugins to be closely related to precompiled script plugins or built-in plugins.  Since the precompiled plugins often have short names and the built-in ones all have short names, I'm mostly convinced that this type of plugin should also have a short name.

It's not something I feel strongly about, though.  If you're unconvinced, I can change it.

Reply:
I don't feel strongly either. But my first impression was "what's that and where is it coming from". Searching only for "library" also takes longer.

Reply:
I will continue to slightly lean in the other direction, but the IDs have now been lengthened.